AddUtil.java

```diff
@@ -16,10 +16,60 @@ public static void myassert(boolean  x) {
     }
     private static int add(int a, int b){
         //Write your code here
+        return (a + b);
     }
 
     private static String add(String a, String b) {
         //Write your code here
+        if((a.charAt(0) == '-') && (b.charAt(0) == '-')) {
+            return "-" + addPvt(a.substring(1), b.substring(1));
+        }
+        else {
+            return addPvt(a, b);
+        }
+    }
+
+        private static String addPvt(String a, String b) {
+        // Handle some base cases
+        if(a.length() == 0){
+            return b;
+        }
+        if(b.length() == 0){
+            return a;
+        }
+        StringBuilder sum = new StringBuilder();
+        int aLen = a.length()-1, bLen = b.length()-1;
```

Comment:
Line 34 also uses a.length(). Better practice, to assign it to a variable if used at  multiple places